src/bcf/report/create_report.rs

```diff
@@ -0,0 +1,295 @@
+                let var_string = String::from("Variant");
+                let var_type: VariantType;
+                let alternatives: Option<String>;
+                let end_position: f64;
+                let plot_start_position;
+
+                match alt {
+                    b"<DEL>" => {
+                        var_type = VariantType::Deletion;
+                        alternatives = None;
+                        end_position = end_pos.unwrap();
+                        plot_start_position = pos as f64 - 0.5;
+                    }
+                    b"<INV>" => {
+                        var_type = VariantType::Inversion;
+                        let rev: String = rfrce.chars().rev().collect();
+                        alternatives = Some(rev.clone());
+                        end_position = end_pos.unwrap();
+                        plot_start_position = pos as f64 - 0.5;
+                    }
+                    b"<DUP>" => {
+                        var_type = VariantType::Duplicate;
+                        let dup: String = [rfrce.clone(), rfrce.clone()].concat();
+                        alternatives = Some(dup.clone());
+                        end_position = end_pos.unwrap();
+                        plot_start_position = pos as f64 - 0.5;
+                    }
+                    _ => {
+                        let mut allel = String::from("");
+
+                        for c in alt {
+                            if *c as char != '<' && *c as char != '>' {
+                                allel.push(*c as char);
+                            }
+                        }
+
+                        match allel.len() {
+                            a if a < rfrce.len() => {
+                                plot_start_position = pos as f64 + 0.5; // start position + 1 due to alignment with deletions from bam (example: ref: ACTT alt: A  -> deletion is just CTT)
+                                end_position = pos as f64 - 0.5 + len as f64;
+                                var_type = VariantType::Deletion;
+                                alternatives = Some(allel.clone());
+                            }
+                            a if a > rfrce.len() => {
+                                plot_start_position = pos as f64;
+                                end_position = pos as f64 + len as f64;
+                                var_type = VariantType::Insertion;
+                                alternatives = Some(allel.clone());
+                            }
+                            _ => {
+                                plot_start_position = pos as f64 - 0.5;
+                                end_position = pos as f64 - 0.5 + len as f64;
+                                var_type = VariantType::Variant;
+                                alternatives = Some(allel.clone());
+                            }
+                        }
+                    }
+                }
+
+                let var = StaticVariant {
```

Comment:
The Static prefix can be removed from all the struct names.

Reply:
Should be fixed with b8be9607f541d707c685bd57391ea7ceac522231. I was able to rename the StaticVariant struct to just Variant. The structs with the Static prefix vary from the original.